src/main/scala/diplomacy/Nodes.scala

```diff
@@ -369,7 +373,11 @@ sealed abstract class MixedNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
       if (iParamsCycleGuard) throw UpwardCycleException()
       iParamsCycleGuard = true
       val i = mapParamsU(iPorts.size, uoParams)
-      require (i.size == iPorts.size, s"Diplomacy error: $context has ${i.size} != ${iPorts.size} up/down inner parameters")
+      require (i.size == iPorts.size,
+        s"""Diplomacy error: $context has ${i.size} != ${iPorts.size} down/up outer parameters:
+           |up-in parameters: [${i.mkString(",")}]
+           |iPorts: [${iPorts.map(_._2.name).mkString(",")}]
```

Comment:
How about:
```
s"""Diplomacy Error: $context produced ${i.size} parameters for ${iPorts.size} inner ports. These sizes must match.
           | Inward parameters produced: [${i.mkString(",")}]
           | Inward ports connected: [${iPorts.map(_._2.name).mkString(",")}]
```

I am not totally clear on whether there is a meaningful distinction between "inner"/"inward"/"input" when talking about ports vs parameters here that it would be useful to include in this message.